packages/lab/src/date-picker/DatePickerOverlay.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  makePrefixer,
+  useFloatingComponent,
+  useForkRef,
+  useFormFieldProps,
+} from "@salt-ds/core";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { clsx } from "clsx";
+import {
+  type ComponentPropsWithoutRef,
+  type ReactNode,
+  forwardRef,
+} from "react";
+import datePickerOverlayCss from "./DatePickerOverlay.css";
+import { useDatePickerOverlay } from "./DatePickerOverlayProvider";
+
+const withBaseName = makePrefixer("saltDatePickerOverlay");
+
+export interface DatePickerOverlayProps
+  extends ComponentPropsWithoutRef<"div"> {
+  children: ReactNode;
+}
+
+export const DatePickerOverlay = forwardRef<
+  HTMLDivElement,
+  DatePickerOverlayProps
+>(function DatePickerOverlay({ className, children, ...rest }, ref) {
+  const targetWindow = useWindow();
+  useComponentCssInjection({
+    testId: "salt-date-picker-overlay",
+    css: datePickerOverlayCss,
+    window: targetWindow,
+  });
+
+  const { Component: FloatingComponent } = useFloatingComponent();
+
+  const { a11yProps } = useFormFieldProps();
+
+  const {
+    state: { open, floatingUIResult },
+    helpers: { getFloatingProps },
+  } = useDatePickerOverlay();
```

Comment:
Not to be fixed in this PR, we need a univeral solution for the user, if they want to explicitly move the position of overlay.

Reply:
@joshwooding  FYI